pkg/kubelet/kubelet_pods_test.go

```diff
@@ -2613,3 +2614,29 @@ func TestGenerateAPIPodStatusHostNetworkPodIPs(t *testing.T) {
 		})
 	}
 }
+
+func TestConvertToAPIContainerStatusesDataRace(t *testing.T) {
+	pod := podWithUIDNameNs("12345", "test-pod", "test-namespace")
+
+	testTimestamp := time.Unix(123456789, 987654321)
+
+	criStatus := &kubecontainer.PodStatus{
+		ID:        pod.UID,
+		Name:      pod.Name,
+		Namespace: pod.Namespace,
+		ContainerStatuses: []*kubecontainer.Status{
+			{Name: "containerA", CreatedAt: testTimestamp},
+			{Name: "containerB", CreatedAt: testTimestamp.Add(1)},
+		},
+	}
+
+	testKubelet := newTestKubelet(t, false)
+	defer testKubelet.Cleanup()
+	kl := testKubelet.kubelet
+
+	for i := 0; i < 2; i++ {
```

Comment:
Would increasing max `i` make it more likely we hit the race (and more likely this test ensures the right behaviour)?

Reply:
My understanding is that the Golang race detector uses pure happens-before detection. As there are no synchronisation events in the tested function (no locks, no channels, ...) each routine should have its own independent vector clock increased independently. As both will attempt to write to the impacted area of memory (`kubecontainer.PodStatus.ContainerStatuses`), I believe the race detector will catch the data race consistently. 

I've tried to reproduce locally by reverting the fix and running `go test ./pkg/kubelet/ -race -run TestConvertToAPIContainerStatusesDataRace` 100 times and it seems to catch the race every time.

Not an expert on the subject, happy to change it or discuss further if you'd like 🙂

Reply:
This explanation matches my (admittedly limited) knowledge and is worth to be distilled in a comment right before the `for` loop because is most likely what any reader in the future is going to wonder. Another option is to just add a link to this github comment.